pkg/apis/crd/v1alpha1/types.go

```diff
@@ -350,14 +350,27 @@ type Rule struct {
 	// +optional
 	Ports []NetworkPolicyPort `json:"ports,omitempty"`
 	// Rule is matched if traffic originates from workloads selected by
-	// this field. If this field is empty, this rule matches all sources.
+	// this field. If this field is empty and this is an ingress rule,
+	// this rule matches all sources.
 	// +optional
 	From []NetworkPolicyPeer `json:"from"`
 	// Rule is matched if traffic is intended for workloads selected by
-	// this field. If this field is empty or missing, this rule matches all
-	// destinations.
+	// this field. This field can't be used with ToService. If this field
+	// and ToService are both empty or missing and this is an egress rule,
+	// this rule matches all destinations.
 	// +optional
 	To []NetworkPolicyPeer `json:"to"`
+	// Rule is matched if traffic is intended for workloads selected by
+	// the Service listed on Service ports. Currently only ClusterIP and
+	// NodePort types Services are supported in this field.
+	// Workloads selected by this field are backend Endpoints+targetPorts
+	// of Services referred in this field. If the ServiceType is NodePort,
+	// workloads selected by this field also include the all Nodes+nodePort.
+	// This field can't be used with To or Ports. If this field and To are
+	// both empty or missing and this is an egress rule, this rule matches
+	// all destinations.
+	// +optional
+	ToService []*ServiceReference `json:"toService"`
```

Comment:
I do not like to define a separate ToService field, but understood it can be complex to validate if we add Service in NetworkPolicyPeer. Maybe we can revisit if later we change AppliedTo to use struct "AppliedTo" and keep NetworkPolicyPeer only for From/To. @abhiraut

Reply:
thanks. i will open up an issue to track this effort, its long over due

Reply:
We already have "FQDN" in "NetworkPolicyPeer" and I think we moved "ToGroup" to "NetworkPolicyPeer" as well. So I think we should stick to this mode?
Besides, "NetworkPolicyPeer" has "Group" which can also refer to a "Service", I think it makes more sense to have the standalone "Service" in "NetworkPolicyPeer" too.

Reply:
Now I see "ToServices" is kind of "To" + "Ports", then please ignore my comment.